docs/architecture/adr/0012-angular-filename-convention.md

```diff
@@ -45,16 +45,27 @@ At Bitwarden we also use a couple of more types:
 - `.request` - Api Request
 - `.response` - Api Response
 - `.type` - Enum
-- `.service.abstraction` - Abstract class for a service, used for DI, not all services needs an
-  abstract class
 
 The class names are expected to use the suffix as part of their class name as well. I.e. a service
 implementation will be named `FolderService`, a request model will be named `FolderRequest`.
 
-In the event a service can't be fully implemented, an abstract class is created with the
-`Abstraction` suffix. This typically happens if the Angular and Node implementations have to differ
-for one reason or another. Traditionally interfaces would be used, but a TypeScript interface cannot
-be used to wire up dependency injection in JavaScript.
+## Abstract & Default Implementations
+
+The Bitwarden clients codebase serves multiple clients, one of which is node based and doesn't use
+the Angular Dependency Injection. In order to make code useable in both Angular and non Angular
+based clients we generally use abstract classes to define the interface. Ideally we would use
+_interfaces_ but TypeScript interfaces cannot be used to wire up dependency injection in JavaScript.
+
+All consumers will use the abstract class as a parameter in their constructor which will be manually
+wired up in the CLI and use Angular dependency injection in the Angular clients. In the case a
+dependency is only used in the Angular client, the abstract class can be omitted and the
```

Comment:
```suggestion
dependency is only used in Angular contexts, the abstract class can be omitted and the
```

We have multiple clients that use Angular, but only only client is *entirely Angular*